### PR TITLE
CRM-20247 Add test & ensure $is_recur is assigned to the message_template.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
---

 * [CRM-20247: $is_recur not always assigned to the template for recurring contributions](https://issues.civicrm.org/jira/browse/CRM-20247)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@seamuslee001 this might affect you? I'm pretty confident in the fix

File: tests/phpunit/api/v3/ContributionTest.php [hunk 1]
```diff
@@ -3490,4 +3515,25 @@ public function testSendMailWithRepeatTransactionAPIFalltoSystemFromNoDefaultFro
...
 
+  /**
+   * Create a Contribution Page with is_email_receipt = TRUE.
+   *
+   * @param array $params
+   *   Params to overwrite with.
+   *
+   * @return array|int
+   */
+  protected function createReceiptableContributionPage($params) {
```
Comment (seamuslee001) [inline tests/phpunit/api/v3/ContributionTest.php]: should set $params = array() here

Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/ContributionTest.php]: yes

Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/ContributionTest.php]: fixed

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I updated in response to your review

**Comment by seamuslee001** [comment]
Looks ok to me, Tests pass and i assume that there are enough tests to prove that its all fine for various receipt types

**Comment by eileenmcnaughton** [comment]
Thanks @seamuslee001

**Pull request merged**